kitt_bar_app/current_batch.rb

```diff
@@ -57,59 +60,59 @@ def fetch_api_data
-  	@ticket_count = @api_data.dig('tickets')&.count || -1
+  	@ticket_count = @tickets&.count || -1
   	@lunch_break  = @api_data.dig('camp', 'on_lunch_break') || false
   end
 
   def batch_open?
-    @api_data && @api_data.dig('status').nil?
+    @api_data && @api_status.nil?
   end
 
   def toggle_duty
     return unless @api_data['on_duties']
 
-    if @api_data.dig('on_duties').map { |on_duty| on_duty['id'] }.include?(@api_data.dig('current_user', 'id'))
-      puts "🟢 On duty"
-      puts "-- 🍕 take a break | #{HttpKitt.patch(@slug, "finish")}"
+    if current_user_in_on_duty?
+      @view.append_with(body: "🟢 On duty")
+      @view.append_with(body: "-- 🍕 take a break", shell: HttpKitt.patch(@slug, "finish"))
     else
-      puts "🔴 Off duty"
-      puts "-- 💻 back to work | #{HttpKitt.post(@slug, "on_duties")}"
+      @view.append_with(body: "🔴 Off duty")
+      @view.append_with(body: "-- 💻 back to work", shell: HttpKitt.post(@slug, "on_duties"))
     end
   end
 
   def tickets
-    if @api_data.dig('tickets').nil? || @api_data.dig('tickets').empty?
-      puts "🎟 #{Color.gray}No tickets yet#{Color.reset}|href=#{tickets_url}|size=12"
+    if @tickets.empty?
+      @view.append_with(body: "🎟 No tickets yet", href: tickets_url, size: 12, color: "gray")
     else
-  	  puts "🎟 #{@api_data.dig('tickets').size} Ticket#{ 's' if @api_data.dig('tickets').size > 1 }|href=#{tickets_url}|size=12"
-      @api_data.dig('tickets').each { |ticket|
-        puts "-- #{ticket.dig('user', 'name')} #{assigned_ticket(ticket)}"
-        # url = "https://kitt.lewagon.com/api/v1/tickets/#{ticket.dig('id')}/take"
-        ticket_info(ticket).each { |line| puts "---- #{line}"}
-        puts "---- #{Color.orange}take it !#{Color.reset} | #{HttpKitt.put(ticket, "take")}" if ticket.dig('policy', 'current_user_can_take')
-        puts "---- #{Color.green}mark as done !#{Color.reset} | #{HttpKitt.put(ticket, "done")}" if ticket.dig('policy', 'current_user_can_mark_as_solved')
-        puts "---- #{Color.red}cancel#{Color.reset} | #{HttpKitt.put(ticket, "cancel")}" if ticket.dig('policy', 'current_user_can_cancel')
+      @view.append_with(body: "🎟 #{@tickets.size} Ticket#{ 's' if @tickets.size > 1 }", href: tickets_url, size: 12)
+      @tickets.each { |ticket|
+        @view.append_with(body: "-- #{ticket.student} #{ticket.assigned_teacher}")
+        @view.append_with(body: "---- #{ticket.header}")
+        ticket.content_formalized.each { |line| @view.append_with(body: "---- #{line}")}
+        @view.append_with(body: "---- take it !", color: 'orange', shell: HttpKitt.put(ticket, "take")) if ticket.current_user_can_take
+        @view.append_with(body: "---- mark as done !", color: 'green', shell: HttpKitt.put(ticket, "done")) if ticket.current_user_can_mark_as_solved
+        @view.append_with(body: "---- cancel", color: 'red', shell: HttpKitt.put(ticket, "cancel")) if ticket.current_user_can_cancel
```

Comment:
Could it be a Ticket instance method?

Reply:
Yep doing this 🤓

Reply:
Also, `Ticket`and `Batch`are still behaving like controllers. Maybe next push should be adding controllers.

Reply:
I guess that would be even better indeed 🙂